bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -305,12 +333,49 @@ protected void addProvider(Provider<E> provider) {
         return null;
     }
 
+    /**
+     * This method traverses over all elements of a provider in the registry and calls the consumer with each element.
+     *
+     * @param provider provider to traverse elements of
+     * @param consumer function to call with element
+     */
+    public void forEach(Provider<E> provider, Consumer<E> consumer) {
```

Comment:
Shouldn't all the new methods have a `protected` modifier instead of a `public` modifier?

Reply:
Makes sense